addon/-private/system/model/model-information-registry.js

```diff
@@ -0,0 +1,226 @@
+import Ember from 'ember';
+import { assert, warn } from "ember-data/-private/debug";
+import EmptyObject from "ember-data/-private/system/empty-object";
+import {
+  typeForRelationshipMeta,
+  relationshipFromMeta
+} from "ember-data/-private/system/relationship-meta";
+
+const {Map, MapWithDefault, get} = Ember;
+
+/**
+ * Contains relationship information that used to live directly on
+ * DS.Model subclasses. The ModelInformationRegistry is used to figure
+ * out inverse relationships for classes and cache that information.
+ *
+ * @private
+*/
+export default Ember.Object.extend({
```

Comment:
it seems like this doesn't need to be an ember object, rather just a plain `function ModelInformation`